Reconstruction/RecTracker/src/components/TrackFit.cpp

```diff
@@ -0,0 +1,207 @@
+#include "TrackFit.h"
+
+#include "DetInterface/ITrackingGeoSvc.h"
+
+#include "datamodel/TrackHitCollection.h"
+
+#include "ACTS/Detector/TrackingGeometry.hpp"
+#include "ACTS/EventData/Measurement.hpp"
+#include "ACTS/Examples/BuildGenericDetector.hpp"
+#include "ACTS/Extrapolation/ExtrapolationCell.hpp"
+#include "ACTS/Extrapolation/ExtrapolationEngine.hpp"
+#include "ACTS/Extrapolation/IExtrapolationEngine.hpp"
+#include "ACTS/Extrapolation/MaterialEffectsEngine.hpp"
+#include "ACTS/Extrapolation/RungeKuttaEngine.hpp"
+#include "ACTS/Extrapolation/StaticEngine.hpp"
+#include "ACTS/Extrapolation/StaticNavigationEngine.hpp"
+#include "ACTS/Fitter/KalmanFitter.hpp"
+#include "ACTS/Fitter/KalmanUpdator.hpp"
+#include "ACTS/MagneticField/ConstantBField.hpp"
+#include "ACTS/Surfaces/PerigeeSurface.hpp"
+#include "ACTS/Utilities/Definitions.hpp"
+#include "ACTS/Utilities/Logger.hpp"
+
+using namespace Acts;
+
+typedef FittableMeasurement<long int> FitMeas_t;
+template <ParID_t... pars>
+using Meas_t = Measurement<long int, pars...>;
+
+struct MyCache {
+  std::unique_ptr<const KF::Step<long int>::JacobianMatrix> jacobian;
+  std::unique_ptr<const BoundParameters> parameters;
+
+  MyCache() = default;
+  MyCache(const MyCache&) = delete;
+  MyCache(MyCache&&) = default;
+};
+
+class MyExtrapolator {
+public:
+  MyExtrapolator(std::shared_ptr<const IExtrapolationEngine> exEngine = nullptr) : m_exEngine(std::move(exEngine)) {}
+
+  MyCache operator()(const FitMeas_t& m, const TrackParameters& tp) const {
+    auto exCell = std::make_unique<ExtrapolationCell<TrackParameters>>(tp);
+    exCell->addConfigurationMode(ExtrapolationMode::Destination);
+    exCell->addConfigurationMode(ExtrapolationMode::FATRAS);
+    exCell->addConfigurationMode(ExtrapolationMode::CollectJacobians);
+    const Surface& sf = getSurface(m);
+
+    m_exEngine->extrapolate(*exCell, &sf);
+    MyCache c;
+    auto j = exCell->extrapolationSteps.back().transportJacobian.release();
+    c.jacobian.reset(new KF::Step<long int>::JacobianMatrix(*j));
+    auto pars = static_cast<const BoundParameters*>(exCell->leadParameters->clone());
+    c.parameters.reset(pars);
+
+    return c;
+  }
+
+private:
+  std::shared_ptr<const IExtrapolationEngine> m_exEngine;
+};
+
+class NoCalibration {
+public:
+  std::unique_ptr<const FitMeas_t> operator()(const FitMeas_t& m, const BoundParameters&) const {
+    return std::make_unique<const FitMeas_t>(m);
+  }
+};
+
+class CacheGenerator {
```

Comment:
Can we move these declarations into the header? Actually, in the coding conventions we wrote down that there should be one header file per class definition. I have no objection to having these small helper classes in the same header, though.